### PR TITLE
Add OLM deployment

### PR DESCRIPTION
**Pull request by meyskens**
Moving the OLM deployment into it's own separate repository

### CODE REVIEW

**Comment by meyskens** [comment]
/hold

need to fix the version numbers

File: README.md [hunk 1]
```diff
@@ -1,2 +1,10 @@
...
+# cert-manager operator deployment
 Definitions for the cert-manager operator published via Red Hat's Operator Lifecycle Manager (OLM)
+
+This repository contains all files that are used to build the [operator](https://operatorhub.io/what-is-an-operator) to deploy cert-manager.
+This is to allow users of OpenShift and OperatorHub to easily install cert-manager into their clusters. It is currently an experimental deployment method.
+This includes the operator itself, based on the Helm operator as well as Dockerfiles to build [UBI](https://connect.redhat.com/about/faq/what-red-hat-universal-base-image-ubi-0) based images.
+
+This repository contains all files that are used by the RedHat image builder to release the operator.
+
+For more info on cert-manager, please see [the cert-manager repository](https://github.com/jetstack/cert-manager) ot [cert-manager.io](https://cert-manager.io )
```
Comment (munnerz) [inline README.md:10]: `ot` -> `or`. Also trailing whitespace after the URL 😄

File: bundle/cert-manager-operator/0.15.0-alpha.0/cert-manager-operator.v0.15.0-alpha.0.clusterserviceversion.yaml [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+kind: ClusterServiceVersion
+metadata:
+  annotations:
+    description: "x509 certificate management for Kubernetes"
+    createdAt: "2020-04-20T12:00:00Z"
+    support: "cert-manager team"
+    alm-examples: ' [{"apiVersion":"operator.cert-manager.io/v1alpha1","kind":"CertManager","metadata":{"name":"cert-manager","namespace":"cert-manager"},"spec":{}}]'
+    capabilities: Full Lifecycle
+    categories: Security
+    certified: 'false'
```
Comment (munnerz) [inline bundle/cert-manager-operator/0.15.0-alpha.0/cert-manager-operator.v0.15.0-alpha.0.clusterserviceversion.yaml:11]: Presumably we flip this somewhere further down the line?

File: bundle/cert-manager-operator/0.15.0-alpha.0/cert-manager-operator.v0.15.0-alpha.0.clusterserviceversion.yaml [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+spec:
+  apiservicedefinitions: {}
+  customresourcedefinitions:
+    owned:
+    - kind: CertManager
+      displayName: CertManager
+      name: certmanagers.operator.cert-manager.io
+      version: v1alpha1
+      description: 'Represents the cert-manager installation'
+  description: 'This is where the description goes'
```
Comment (munnerz) [inline bundle/cert-manager-operator/0.15.0-alpha.0/cert-manager-operator.v0.15.0-alpha.0.clusterserviceversion.yaml:24]: nit: maybe take first paragraph from the website?

File: bundle/cert-manager-operator/0.15.0-alpha.0/cert-manager-operator.v0.15.0-alpha.0.clusterserviceversion.yaml [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+    spec:
+      clusterPermissions:
+      - serviceAccountName: cert-manager-operator
+        rules:
+        - apiGroups:
+          - '*'
+          resources:
+          - '*'
+          verbs:
+          - '*'
```
Comment (munnerz) [inline bundle/cert-manager-operator/0.15.0-alpha.0/cert-manager-operator.v0.15.0-alpha.0.clusterserviceversion.yaml:39]: We really should look to reduce this at some point.. for now though LGTM :)

File: bundle/cert-manager-operator/0.15.0-alpha.0/cert-manager-operator.v0.15.0-alpha.0.clusterserviceversion.yaml [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+  - certificates
+  links:
+  - name: repository
+    url: https://github.com/jetstack/cert-manager
+  - name: conatinerImage
+    url: https://quay.io/jetstack/cert-manager:latest
+  maintainers:
+  - email: cert-manager-maintainers@jetstack.io
+    name: cert-manager team
+  maturity: alpha
```
Comment (munnerz) [inline bundle/cert-manager-operator/0.15.0-alpha.0/cert-manager-operator.v0.15.0-alpha.0.clusterserviceversion.yaml:87]: What does maturity here refer to? When does this change?

File: bundle/cert-manager-operator/0.15.0-alpha.0/cert-manager-operator.v0.15.0-alpha.0.clusterserviceversion.yaml (hunk 3 above)
Comment (meyskens) [reply bundle/cert-manager-operator/0.15.0-alpha.0/cert-manager-operator.v0.15.0-alpha.0.clusterserviceversion.yaml:24]: whoops...

File: bundle/cert-manager-operator/0.15.0-alpha.0/cert-manager-operator.v0.15.0-alpha.0.clusterserviceversion.yaml (hunk 2 above)
Comment (meyskens) [reply bundle/cert-manager-operator/0.15.0-alpha.0/cert-manager-operator.v0.15.0-alpha.0.clusterserviceversion.yaml:11]: We should, for now it isn't fully certified yet

File: bundle/cert-manager-operator/0.15.0-alpha.0/cert-manager-operator.v0.15.0-alpha.0.clusterserviceversion.yaml (hunk 4 above)
Comment (meyskens) [reply bundle/cert-manager-operator/0.15.0-alpha.0/cert-manager-operator.v0.15.0-alpha.0.clusterserviceversion.yaml:39]: Agreed https://github.com/jetstack/cert-manager-olm/issues/3

File: bundle/cert-manager-operator/0.15.0-alpha.0/cert-manager-operator.v0.15.0-alpha.0.clusterserviceversion.yaml (hunk 5 above)
Comment (meyskens) [reply bundle/cert-manager-operator/0.15.0-alpha.0/cert-manager-operator.v0.15.0-alpha.0.clusterserviceversion.yaml:87]: This should refer to the application release, which for now is Alpha

**Comment by meyskens** [comment]
@munnerz ready for re-review

**Pull request merged**